src/d3a/models/market/one_sided.py

```diff
@@ -119,13 +119,13 @@ def accept_offer(self, offer_or_id: Union[str, Offer], buyer: str, *, energy: in
         if trade_rate is None:
             trade_rate = offer.price / offer.energy
 
-        energy_portion = energy / offer.energy
         self._sorted_offers = sorted(self.offers.values(),
                                      key=lambda o: o.price / o.energy)
         try:
             if time is None:
                 time = self._now
             if energy is not None:
+                energy_portion = energy / offer.energy
```

Comment:
This is just refactoring, right? The error was due to the deleted ChainTradeException right?

Reply:
Actually this was the reason why the integration tests were failing just before this PR. It was introduced during a last careless refactoring before I opened this PR, which introduced moving this calculation outside of 'energy is not None' block. This had the effect that at some points, energy was indeed None, which crashed the whole simulation. That's why I fixed that and moved it back inside the if block.

Reply:
Ah, ok! Thanks!